src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -1104,7 +1104,7 @@ public void onWorldTick(@NotNull final TickEvent.WorldTickEvent event)
             //  Cleanup disappeared citizens
             //  It would be really nice if we didn't have to do this... but Citizens can disappear without dying!
             //  Every CITIZEN_CLEANUP_TICK_INCREMENT, cleanup any 'lost' citizens
-            if ((event.world.getWorldTime() % CITIZEN_CLEANUP_TICK_INCREMENT) == 0 && areAllColonyChunksLoaded(event) && townHall != null)
+            if ((event.world.getTotalWorldTime() % CITIZEN_CLEANUP_TICK_INCREMENT) == 0 && areAllColonyChunksLoaded(event) && townHall != null)
```

Comment:
We could improve server performance quite a bit by making a method that gives us total world time with a random offset unique to each colony (colonyID perhaps). That way this stuff wont all get called the same tick for every colony.